### PR TITLE
Task2

### PR DESCRIPTION
**Pull request by VasilMitovClouway**

### CODE REVIEW

**Comment by freekman** [comment]
add me a description what have you done in this PR

**Comment by VasilMitovClouway** [comment]
Made a project, that uses last task as a module named codec for the module message preview serving task2. Changed the codec for Json from using Gson to using Jackson. The main class named App takes the args[] from psvm on startup and unmarshalls the file message.json from the task requirement to print the average stats of user count and the average years.

File: bigFile.json [hunk 1]
```diff
@@ -0,0 +1,329 @@
+{
```
Comment (freekman) [inline bigFile.json]: remove this .json also the big .xml

Comment (VasilMitovClouway) [inline bigFile.json]: Done

**Comment by freekman** [comment]
Here is yours 
screenshot from 2016-07-18 10 52 30

Here is an example with title and description for a PR
screenshot from 2016-07-18 10 51 10

**Comment by VasilMitovClouway** [comment]
I will have that in mind next time.

File: messagepreview/src/main/java/com.clouway/Operation.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import javax.xml.bind.JAXBException;
+import java.io.FileNotFoundException;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * @author Vasil Mitov (v.mitov.clouway@gmail.com)
+ */
+public class Operation {
+  String[] arguments;
```
Comment (freekman) [inline messagepreview/src/main/java/com.clouway/Operation.java]: why do you forget  the modifiers

File: messagepreview/src/main/java/com.clouway/Operation.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  public Operation(String[] arguments) {
+    this.arguments = arguments;
+  }
+
+  public String compile() throws FileNotFoundException, JAXBException {
+    String operationResult = "";
+    String extention = "";
+    XmlCodec xmlCodec = new XmlCodec(new ListOfObjects(), new User());
+    JsonCodec jsonCodec = new JsonCodec();
+    if (arguments[0].equals("-type")) {
```
Comment (freekman) [inline messagepreview/src/main/java/com.clouway/Operation.java]: extract the rguments[0],.... in variables to enhance the readability

File: messagepreview/src/main/java/com.clouway/Operation.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+ */
+public class Operation {
+  String[] arguments;
+  Codec codec = new Codec();
+
+  public Operation(String[] arguments) {
+    this.arguments = arguments;
+  }
+
+  public String compile() throws FileNotFoundException, JAXBException {
```
Comment (freekman) [inline messagepreview/src/main/java/com.clouway/Operation.java]: i would like this to compile ```json``` also

Comment (freekman) [inline messagepreview/src/main/java/com.clouway/Operation.java]: add tests for this class

File: messagepreview/src/main/java/com.clouway/User.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
+package com.clouway;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * @author Vasil Mitov (v.mitov.clouway@gmail.com)
+ */
+public class User {
+  String firstname;
```
Comment (freekman) [inline messagepreview/src/main/java/com.clouway/User.java]: .

File: messagepreview/src/main/java/com.clouway/Codec.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+
+/**
+ * @author Vasil Mitov (v.mitov.clouway@gmail.com)
+ */
+public class Codec {
```
Comment (freekman) [inline messagepreview/src/main/java/com.clouway/Codec.java]: i find this class Codec very confusing .And here is why you have JsonCodec and XmlCodec who are used int the Codec class , if you change something in eater of the classes you will have to change the codec class.
Make thing little more simple 
```js
one interface MessageCodec
and two classes that imp. the MessageCodec 
```

**Comment by freekman** [comment]
Don't write tests for the parser's i have seen them in the previous task

File: codec/src/main/java/com/clouway/Codec.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+import javax.xml.bind.JAXBException;
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * @author Vasil Mitov (v.mitov.clouway@gmail.com)
+ */
+public interface Codec {
+  String marshall(Object object) throws IOException, JAXBException;
```
Comment (freekman) [inline codec/src/main/java/com/clouway/Codec.java]: remove the ``` throws Exception```

File: codec/src/main/java/com/clouway/Codec.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+  <T> T unmarshall(final TypeReference<T> type, InputStream stream) throws IOException, JAXBException;
```
Comment (freekman) [inline codec/src/main/java/com/clouway/Codec.java]: .

**Comment by freekman** [comment]
is this PR ready for review

**Comment by VasilMitovClouway** [comment]
No im still doing the changes from yesterday.

**Comment by VasilMitovClouway** [comment]
The task is ready.

**Pull request merged**